File: apps/electorium_web/assets/src/models/Candidate.js
import { Model } from '@vuex-orm/core';
import Student from './Student';

export default class Candidate extends Model {

    static entity = 'candidates';

    static fields () {

        return {

            id: this.uid(),
            nomination: this.attr(''),
            proposal: this.attr(''),
            photo: this.attr(''),
            student_id: this.attr(''),
            student: this.hasOne(Student, 'student_id')
        };
    }
}

File: apps/electorium/lib/electorium/students/candidate.ex
defmodule Electorium.Students.Candidate do
  use Ecto.Schema
  import Ecto.Changeset

  alias Electorium.Students.Student

  @derive {Jason.Encoder, only: [:nomination,
                                 :proposal,
                                 :photo
                                ]}

  schema "candidates" do
    field :nomination, :string
    field :proposal, :string
    field :photo, :string
    belongs_to :student, Student

    timestamps()
  end

  @doc false
  def changeset(candidate, attrs \\ %{}) do
    candidate
    |> cast(attrs, [:nomination,
                   :proposal,
                   :photo])
    |> validate_required([:nomination,
                         :proposal,
                         :photo])
    |> unique_constraint(:student_id)
  end
end
